test/functional/legacy/061_undo_tree_spec.lua

```diff
@@ -0,0 +1,280 @@
+-- Tests for undo tree and :earlier and :later.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source, eq, eval, clear, execute, expect, wait =
+  helpers.feed, helpers.insert, helpers.source, helpers.eq, helpers.eval,
+  helpers.clear, helpers.execute, helpers.expect, helpers.wait
+
+local function expect_empty_buffer()
+  -- The space will be removed by helpers.dedent but is needed as dedent will
+  -- throw an error if it can not find the common indent of the given lines.
+  return expect(' ')
+end
+local function expect_line(line)
+  return eq(line, eval('getline(".")'))
+end
+local function write_file(name, text)
+  local file = io.open(name, 'w')
+  file:write(text)
+  file:flush()
+  file:close()
+end
+
+describe('undo:', function()
+  before_each(clear)
+  teardown(function()
+    os.remove('Xtest.source')
+  end)
+
+  describe(':earlier and :later', function()
+    before_each(function()
+      os.remove('Xtest')
+    end)
+    teardown(function()
+      os.remove('Xtest')
+    end)
+
+    it('work with time specifications and g- and g+', function()
+      -- We write the test text to a file in order to prevent nvim to record
+      -- the inserting of the text into the undo history.
+      write_file('Xtest', '\n123456789\n')
+
+      -- Note: This test is not executed directly like the other tests.  It is
+      -- encapsuled in a function and will be tried up to three times if it
+      -- fails.  This is done because the test is dependent on wall clock time
+      -- and somtimes fails if the system is under load or the test is paused
+      -- for a long enough interval.
```

Comment:
Instead of this comment, consider renaming `test()` to `test_earlier_later()` with this comment:

```
-- `:earlier` and `:later` are (obviously) time-sensitive, so this test
-- sometimes fails if the system is under load. It is wrapped in a local 
-- function to allow multiple attempts.
```